src/main/java/se/simonsoft/cms/transform/service/TransformServiceXsl.java

```diff
@@ -0,0 +1,289 @@
+		try {
+			
+			final InputStream transformStream = checkStreamIsNotEmpty(streamProvider.get());
+			boolean pathExists = pathExists(patchset.getRepository(), relPath);
+			if (!pathExists) {
+				logger.debug("No file at path: '{}' will add new file.", relPath);
+				FileAdd fileAdd = new FileAdd(relPath, transformStream);
+				fileAdd.setPropertyChange(properties);
+				patchset.add(fileAdd);
+				addFolderExists(patchset, relPath.getParent());
+			} else if (overwrite){
+				logger.debug("Overwrite is allowed, existing file at path '{}' will be modified.", relPath.getPath());
+				CmsItemId itemId = new CmsItemIdArg(patchset.getRepository(), relPath);
+				CmsItemLockCollection locks = commit.lock(TRANSFORM_LOCK_COMMENT, patchset.getBaseRevision(), itemId.getRelPath());
+				if (locks != null && locks.getSingle() == null) {
+					throw new IllegalStateException("Unable to retrieve the lock token after locking " + itemId);
+				}
+				patchset.addLock(locks.getSingle());
+				FileModificationLocked fileMod = new FileModificationLocked(relPath, transformStream);
+				fileMod.setPropertyChange(properties);
+				patchset.add(fileMod);
+			} else {
+				throw new IllegalStateException("Item already exists, config prohibiting overwrite of existing items.");
+			}
+			
+		} catch (IOException e) {
+			throw new RuntimeException("Failed to read stream from transform.", e);
+		} catch (EmptyStreamException e) {
+			logger.warn("Transform of item at path: '{}'  resulted in empty document, will be discarded.", relPath);
+		}
+	}
+	
+	private CmsItemPath getOutputPath(CmsItemId itemId, String output) {
+		
+		CmsItemPath pathResult;
+		
+		if (output == null || output.trim().isEmpty()) {
+			pathResult = new CmsItemIdArg(itemId.getRepository(), itemId.getRelPath().getParent()).getRelPath();
+			logger.debug("Output folder is not specified will default to items parent folder.");
+		} else {
+			pathResult = new CmsItemIdArg(itemId.getRepository(), new CmsItemPath(output)).getRelPath();
+			logger.debug("Output folder is specified: {}", output);
+		}
+		
+		return pathResult;
+	}
+
+	private TransformerService getTransformerService(CmsItemId itemId, String stylesheet) {
+		
+		TransformerService resultService;
+		
+		if (stylesheet.startsWith("/")) {
+			CmsItemId styleSheetItemId = new CmsItemIdArg(itemId.getRepository(), new CmsItemPath(stylesheet));
+			logger.debug("Using stylesheet from CMS: {}", styleSheetItemId.getLogicalId());
+			CmsItem styleSheetItem = itemLookup.getItem(styleSheetItemId);
+			
+			ByteArrayOutputStream baos = new ByteArrayOutputStream();
+			styleSheetItem.getContents(baos);
+			resultService = transformerServiceFactory.buildTransformerService(new StreamSource(IOUtils.toInputStream(baos.toString(StandardCharsets.UTF_8), StandardCharsets.UTF_8)));
+		} else {
+			logger.debug("Using CMS built in stylesheet: {}", stylesheet);
+			resultService = stylesheets.get(stylesheet); 
```

Comment:
What if it is unknown?

Reply:
Will throw IAE from getTransformerService(); if the stylesheet is uknown. Added test coverage.